_posts/2018-05-24-kotlin-project-setup.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: Kotlin Gradle projects and gRPC
+published: false
+permalink: blog/kotlin-gradle-projects
+author: Spencer Fang
+author-link: https://github.com/zpencer
+company: Google
+company-link: https://www.google.com
+---
+
+# Kotlin Gradle projects and gRPC
+
+Did you know that gRPC Java now has out of box support for Kotlin projects built with Gradle? Kotlin is a modern, statically typed language developed by JetBrains that targets the JVM and Android. It is generally easy for Kotlin programs to interoperate with existing Java libraries. We have added support to the [protobuf-gradle-plugin](https://github.com/google/protobuf-gradle-plugin/releases) so that the generated Java libraries are automatically picked up by Kotlin. You can now add the protobuf-gradle-plugin to your Kotlin project, and use gRPC just like you would with a typical Java project.
+
+<!--more-->
+The follow examples show you how to configure a project for a JVM application and an Android application using Kotlin.
+
+### Kotlin gRPC client and server
+
+The full example can be found [here](https://github.com/grpc/grpc-java/tree/master/examples/example-kotlin).
+
+Configuring gRPC for a Kotlin project is the same as configuring it for a Java project.
+
+Below is a snippet of the example project's `build.gradle` highlighting some Kotlin related sections:
+```groovy
+apply plugin: 'kotlin'
+apply plugin: 'com.google.protobuf'
+
+// Generate IntelliJ IDEA's .idea & .iml project files.
+// protobuf-gradle-plugin automatically registers *.proto and the gen output files
+// to IntelliJ as sources.
+// For best results, install the Protobuf and Kotlin plugins for IntelliJ.
+apply plugin: 'idea'
+
+buildscript {
+  ext.kotlin_version = '1.2.21'
+
+  repositories {
+    mavenCentral()
+  }
+  dependencies {
+    classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.5'
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+  }
+}
+
+dependencies {
+  compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
+  // The rest of the projects dep are added below, refer to example URL
+}
+
+protobuf {
+  // The normal gRPC configuration goes here, refer to example URL
```

Comment:
if this information is small enough, can we reproduce it here?

Reply:
Done